gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -293,15 +293,11 @@ public void submitFlowToDagManager(FlowSpec flowSpec) throws IOException, Interr
   public void submitFlowToDagManager(FlowSpec flowSpec, Dag<JobExecutionPlan> jobExecutionPlanDag)
       throws IOException {
     try {
-      // Send the dag to the DagManager
-      this.dagManager.addDag(jobExecutionPlanDag, true, true);
-
-      /*
-      Adhoc flows can be deleted after persisting it in DagManager as the DagManager's failure recovery method ensures
-      it will be executed in the event of downtime. Note that the responsibility of the multi-active scheduler mode ends
-      after this method is completed AND the consumption of a launch type event is committed to the consumer.
+      /* Send the dag to the DagManager
+      Note that the responsibility of the multi-active scheduler mode ends after this method is completed AND the
+      consumption of a launch type event is committed to the consumer.
        */
-      deleteSpecFromCatalogIfAdhoc(flowSpec);
+      this.dagManager.addDagAndRemoveAdhocFlowSpec(flowSpec, jobExecutionPlanDag, true, true);
```

Comment:
will this be called on all participants, whether or not they are active?

Reply:
Yes this will be called by all participants. At present we want to delete the flowSpec only after the active dagManager handles it. In MA execution, we'd delete it after lease arbitration and lease completion.